src/StrawberryShake/Client/src/Transport.WebSockets/Protocols/GraphQlWs/SubscriptionTransportWsProtocol.cs

```diff
@@ -0,0 +1,155 @@
+            }
+
+            var writer = new SocketMessageWriter();
+
+            writer.WriteStartObject();
+            writer.WriteType(MessageTypes.Subscription.Start);
+            writer.WriteId(operationId);
+            writer.WriteStartPayload();
+            _serializer.Serialize(request, writer);
+            writer.WriteEndObject();
+
+            await client
+                .SendAsync(writer.Body, cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        public async Task StopOperationAsync(
+            string operationId,
+            CancellationToken cancellationToken)
+        {
+            EnsureInitialized(out ISocketClient client);
+
+            if (client.IsClosed)
+            {
+                return;
+            }
+
+            var writer = new SocketMessageWriter();
+
+            writer.WriteStartObject();
+            writer.WriteType(MessageTypes.Subscription.Stop);
+            writer.WriteId(operationId);
+            writer.WriteEndObject();
+
+            await client.SendAsync(writer.Body, cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        public async Task InitializeAsync(
+            ISocketClient socketClient,
+            CancellationToken cancellationToken)
+        {
+            _socketClient = socketClient;
+            var messageWriter = new SocketMessageWriter();
+            messageWriter.WriteStartObject();
+            messageWriter.WriteType(MessageTypes.Connection.Initialize);
+            messageWriter.WriteEndObject();
+
+            await _socketClient
+                .SendAsync(messageWriter.Body, cancellationToken)
+                .ConfigureAwait(false);
+
+            await _handler.OnConnectAsync(socketClient).ConfigureAwait(false);
+        }
+
+        public async Task TerminateAsync(CancellationToken cancellationToken)
+        {
+            EnsureInitialized(out ISocketClient client);
+
+            var messageWriter = new SocketMessageWriter();
+            messageWriter.WriteStartObject();
+            messageWriter.WriteType(MessageTypes.Connection.Terminate);
+            messageWriter.WriteEndObject();
+
+            await _handler.OnDisconnectAsync(client).ConfigureAwait(false);
+
+            await client.SendAsync(messageWriter.Body, cancellationToken).ConfigureAwait(false);
+
+            _socketClient = null;
+        }
+
+        public void Subscribe(OnReceiveAsync listener)
+        {
+            _listeners.Add(listener);
+        }
+
+        public void Unsubscribe(OnReceiveAsync listener)
+        {
+            _listeners.Remove(listener);
+        }
+
+        public async ValueTask Notify(
+            string messageId,
+            JsonDocument document,
+            CancellationToken cancellationToken)
+        {
+            foreach (var listener in _listeners)
+            {
+                await listener(messageId, document, cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
I would prefere to not have this observable pattern here, i needed it to connect the protocol to the operation manager.